programs/bpf_loader/src/syscalls.rs

```diff
@@ -88,147 +88,139 @@ impl SyscallConsume for Rc<RefCell<dyn ComputeMeter>> {
 /// Simple bump allocator, never frees
 use crate::allocator_bump::BPFAllocator;
 
-/// Default program heap size, allocators
-/// are expected to enforce this
-const DEFAULT_HEAP_SIZE: usize = 32 * 1024;
-
 pub fn register_syscalls<'a>(
     loader_id: &'a Pubkey,
-    vm: &mut EbpfVm<'a, BPFError>,
+    vm: &mut EbpfVm<'a, BPFError, crate::ThisInstructionMeter>,
     callers_keyed_accounts: &'a [KeyedAccount<'a>],
     invoke_context: &'a mut dyn InvokeContext,
-) -> Result<MemoryRegion, EbpfError<BPFError>> {
+    heap: Vec<u8>,
+) -> Result<(), EbpfError<BPFError>> {
     let compute_budget = invoke_context.get_compute_budget();
 
     // Syscall functions common across languages
 
-    vm.register_syscall_ex("abort", syscall_abort)?;
-    vm.register_syscall_with_context_ex("sol_panic_", Box::new(SyscallPanic { loader_id }))?;
-    vm.register_syscall_with_context_ex(
-        "sol_log_",
-        Box::new(SyscallLog {
+    vm.register_syscall(hash_symbol_name(b"abort"), Syscall::Function(syscall_abort))?;
```

Comment:
Why expose `hash_sybmol_name` up to this level?

Reply:
That has to do with the refactoring of syscall functions and objects, how they are registered and bound. It was easier this way than to have a separate method for all combinations of these. But we can try to turn it back, once the JIT caching refactoring is done.

Reply:
Looked like all the calls required a call to `hash_symbol_name`, that could just be done internally and pass a `&str` instead?

Reply:
Refactoring is done, and register by name is back: https://github.com/solana-labs/rbpf/pull/107